flink-ml-lib/src/main/java/org/apache/flink/ml/common/gbt/GBTModelParams.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.gbt;
+
+import org.apache.flink.ml.classification.gbtclassifier.GBTClassifierModel;
+import org.apache.flink.ml.common.param.HasCategoricalCols;
+import org.apache.flink.ml.common.param.HasFeaturesCol;
+import org.apache.flink.ml.common.param.HasLabelCol;
+import org.apache.flink.ml.common.param.HasPredictionCol;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.StringArrayParam;
+import org.apache.flink.ml.regression.gbtregressor.GBTRegressorModel;
+
+/**
+ * Params of {@link GBTClassifierModel} and {@link GBTRegressorModel}.
+ *
+ * <p>If the input features come from 1 column of vector type, `featuresCol` should be used, and all
+ * features are treated as continuous features. Otherwise, `inputCols` should be used for multiple
+ * columns. Columns whose names specified in `categoricalCols` are treated as categorical features,
+ * while others are continuous features.
+ *
+ * <p>NOTE: `inputCols` and `featuresCol` are in conflict with each other, so they should not be set
+ * at the same time. In addition, `inputCols` has a higher precedence than `featuresCol`, that is,
+ * `featuresCol` is ignored when `inputCols` is not `null`.
+ *
+ * @param <T> The class type of this instance.
+ */
+public interface GBTModelParams<T>
+        extends HasFeaturesCol<T>, HasLabelCol<T>, HasCategoricalCols<T>, HasPredictionCol<T> {
```

Comment:
It is a little complicated. I'll answer your questions one by one.

1. A more complete parameter list is in `BaseGBTParams`. `GBTModelParams` contains only parameters for the GBT models.

2. Generally, we support all algorithm-specific parameters from SparkML exception `maxMemoryInMB`, `cacheNodeIds`, and `checkpointInterval`.  (Note some parameters are added but corresponding functions are not implemented yet in this PR.) 

3. `HasVarianceImpurity` is not supported because SparkML actually does not use it [1]. But we may support it in the future.  `HasWeightCol` is added in the `BaseGBTParams` but corresponding function is not implemented now.

4. `HasCategoricalCols` is needed when multiple columns are used as input to help distinguish between numerical and categorical values.


[1] https://github.com/apache/spark/blob/27ed89b7be5ebb91e4a0b106b1669a7867a6012d/mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala#L118

Reply:
I've update the description of this PR with details about supported and unsupported features.